notebooks/cookbook/Enhanced sampling methods/Umbrella sampling MD.ipynb

```diff
@@ -0,0 +1,436 @@
+    "        self.addComputePerDof(\"sigma\", \"sqrt(kT/m)\")\n",
+    "        self.addGlobalVariable(\"a\", np.exp(-1 * gamma)) #vscale?\n",
+    "        self.addGlobalVariable(\"b\", np.sqrt(1 - np.exp(-2 * gamma))) # noise-scale?\n",
+    "\n",
+    "        #before position restraints\n",
+    "        #self.addPerDofVariable(\"oldx\", 0)\n",
+    "\n",
+    "        # Compute steps for Langevin - attempt at VRORV splitting?\n",
+    "        self.addComputePerDof(\"v\", \"v + (dt / 2) * f/m\")\n",
+    "        # langevin like normal\n",
+    "        self.addComputePerDof(\"x\", \"x + (dt / 2)*v\")\n",
+    "        self.addComputePerDof(\"oldx\", \"x\")\n",
+    "        self.addConstrainPositions()\n",
+    "        self.addComputePerDof(\"v\", \"v + (x - oldx)/(dt / 2)\")\n",
+    "        self.addComputePerDof(\"v\", \"(a * v) + (b * sigma * gaussian)\") # taken from openmm-tutorial on custom integrators (link 1 above)\n",
+    "        self.addComputePerDof(\"x\", \"x + (dt / 2)*v\")\n",
+    "        self.addComputePerDof(\"oldx\", \"x\")\n",
+    "        self.addConstrainPositions()\n",
+    "        self.addComputePerDof(\"v\", \"v + (x - oldx) / (dt / 2)\")\n",
+    "        # Use the same line as what you calculated above\n",
+    "        self.addComputePerDof(\"v\", \"v + (dt / 2) * f/m\")\n",
+    "        # compute the V and deltaV values\n",
+    "        self.addComputeGlobal(\"V0\", \"energy\")\n",
+    "        self.addConstrainVelocities()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "integrator = LangevinEqIntegrator(dt=time_step,\n",
+    "                                  temperature=300*kelvin,\n",
+    "                                  collision_rate=1.0/picosecond)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create the simulation object"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "platform = Platform.getPlatformByName('OpenCL')\n",
+    "platform.setPropertyDefaultValue('Precision', 'mixed')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "simulation = Simulation(pdb.topology,system, integrator, platform)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "simulation.reporters.append(DCDReporter('./'+current_file+''+ '.dcd', restart_freq))\n",
+    "simulation.reporters.append(CheckpointReporter('./'+current_file+''+ '.chk', min(prd_steps, 10*restart_freq)))\n",
+    "simulation.reporters.append(StateDataReporter(open('./log.' + current_file+'', 'w'), log_freq, step=True, potentialEnergy=True, kineticEnergy=True, totalEnergy=True, temperature=True, volume=True, density=True, speed=True))"
```

Comment:
This is unnecessary.  Unless you specifically refer to the output of a reporter later on, it's relevant to what you're teaching.